core/core.go

```diff
@@ -303,6 +317,44 @@ func (c *Core) addToQueueIfNotAppended(block *types.Block) {
 	}
 }
 
+// SetSyncTarget sets the sync target entropy based on the prime blocks
+func (c *Core) SetSyncTarget(header *types.Header) {
+	if c.sl.subClients == nil || header.Hash() == c.sl.config.GenesisHash {
+		return
+	}
+
+	// Check if the header is in the badSyncTargets cache
+	_, ok := c.badSyncTargets.Get(header.Hash())
+	if ok {
+		return
+	}
+
+	nodeCtx := common.NodeLocation.Context()
+	// Set Sync Target for subs
+	if nodeCtx != common.ZONE_CTX {
+		if header != nil {
+			for i := range c.sl.subClients {
+				if c.sl.subClients[i] != nil {
+					c.sl.subClients[i].SetSyncTarget(context.Background(), header)
+				}
+			}
+		}
+	}
+	if c.syncTarget == nil || c.syncTarget.ParentEntropy().Cmp(header.ParentEntropy()) < 0 {
+		c.syncTarget = header
```

Comment:
If I'm reading this correctly, we always set the sub sync target to `header`, but we may not set our sync target to that same header if the entropy check doesn't pass. Why would we want this to happen? Should the entropy check happen before setting sub sync target?

Reply:
yeah changed it, decision is done in Prime and we just set in subs